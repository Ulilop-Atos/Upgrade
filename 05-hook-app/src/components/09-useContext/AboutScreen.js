import { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);

  const handleClick = () => {
    setUser({});
  };

  return (
    <>
      <h1>AboutScreen</h1>
      <hr />
      <pre className="container">
        {JSON.stringify(user,null,2)}
      </pre>

      <button
        className="btn btn-primary"
        onClick={handleClick}
      >
        Logout
      </button>
    </>
  )
}

export default AboutScreen;