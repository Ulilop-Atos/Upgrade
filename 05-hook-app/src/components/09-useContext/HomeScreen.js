import { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {

  const { user } = useContext(UserContext);

  return (
    <>
      <h1>HomeScreen</h1>
      <hr />
      <pre className="container">
        {JSON.stringify(user,null,2)}
      </pre>
    </>
  )
}

export default HomeScreen;