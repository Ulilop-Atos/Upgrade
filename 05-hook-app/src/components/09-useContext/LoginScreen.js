import { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {

  const {setUser} = useContext(UserContext);

  const userExample = {
    id: 123,
    name: 'Ferdinand',
    email: 'ferdinand@mail.com',
  };

  return (
    <>
      <h1>LoginScreen</h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={() => setUser(userExample)}
      >
        Login
      </button>
    </>
  )
}

export default LoginScreen;