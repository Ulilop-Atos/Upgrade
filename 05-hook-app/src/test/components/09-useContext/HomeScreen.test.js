import { mount } from "enzyme";
import HomeScreen from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Tests on <HomeScreen />', () => { 
  
  const user = {
    id: 0, 
    name: 'Test Name',
    email: 'test@mail.com',
  }

  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider value={{user, setUser}}>
      <HomeScreen />
    </UserContext.Provider>
  );
  
  test('should display properly', () => { 
    expect(wrapper).toMatchSnapshot();
   })
 })

