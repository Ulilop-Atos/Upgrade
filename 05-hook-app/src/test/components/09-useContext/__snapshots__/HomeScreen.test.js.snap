// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <HomeScreen /> should display properly 1`] = `
Array [
  <h1>
    HomeScreen
  </h1>,
  <hr />,
  <pre
    className="container"
  >
    {
  "id": 0,
  "name": "Test Name",
  "email": "test@mail.com"
}
  </pre>,
]
`;
